src/panels/config/entities/ha-config-entities.ts

```diff
@@ -206,14 +207,14 @@ export class HaConfigEntities extends SubscribeMixin(LitElement) {
 
   public disconnectedCallback() {
     super.disconnectedCallback();
-    const dialog = document
-      .querySelector("home-assistant")!
-      .shadowRoot!.querySelector("dialog-entity-registry-detail") as
-      | DialogEntityRegistryDetail
-      | undefined;
-    if (dialog) {
-      dialog.closeDialog();
+    if (!this.getDialog) {
+      return;
+    }
+    const dialog = this.getDialog();
+    if (!dialog) {
+      return;
     }
+    dialog.closeDialog();
```

Comment:
I think that we should just return a `closeDialog` function from the show function instead of a getDialog function. By exposing getDialog, the user still needs to know about the implementation details? _maybe_ that's ok… I dunno.